### ISSUE TITLE
Test and Address cases for Action nodes

### ISSUE THREAD

**Issue by palashborhanuddin** [palashborhanuddin/RefactoringMatcher#8]
1. Type determine for the statement as "Collections.emptyMap()" or "Integer.valueOf()"
2. Use method return type as Class type for the chain of method invocation such as "memcachedWriteInstancesByServerGroup.get(key).get(0)"
3. Ignore the Element or Key type for the collection types? such as, for the node "Map<ServerGroup,List<EVCacheClient>>.get"

**Comment by tsantalis** [comment]
Using APIFinder `Collections.emptyMap()` should have a return type `Map<K,V>`

if `return memcachedWriteInstancesByServerGroup.get(key).get(0)` then you can use the method return type.

For the sake of having more Generic groums, we can ignore Element and/or Key types for collections.

**Comment by palashborhanuddin** [comment]
> * Type determine for the statement as "Collections.emptyMap()" or "Integer.valueOf()"
> * Use method return type as Class type for the chain of method invocation such as "memcachedWriteInstancesByServerGroup.get(key).get(0)"
> * Ignore the Element or Key type for the collection types? such as, for the node "Map<ServerGroup,List>.get"

Done for the above cases. 
- correct type determination for methodInvocation node
- ignore Element and/or Key types for collections for action nodes
- Use method return type as Class type for the following method in chain of method invocation
- use invoker type filter for TypeInferenceFluentAPI of JarAnalyser
- addressed few other cases for action nodes during development/review

Target test code:
```JAVA
public List<TryStatementObject> getTryStatements() {
        List<TryStatementObject> tryStatements = new ArrayList<TryStatementObject>();
        int size = tryStatements.size();
        int size = this.statementList.size();
        Map<String, String> emptymap = Collections.emptyMap();
        Integer i = java.lang.Integer.valueOf(1).toString().equals("1");
        String s = String.valueOf(1);
        if(this.getType().equals(StatementType.TRY))
            tryStatements.add((TryStatementObject)this);
        for(AbstractStatement statement : statementList) {
            if(statement instanceof CompositeStatementObject) {
                CompositeStatementObject composite = (CompositeStatementObject)statement;
                tryStatements.addAll(composite.getTryStatements());
            }
        }
        return tryStatements;
    }
```
GROUM representation:
```JAVA
Groum{
nodes=[1 ArrayList.<init>, 2 List.size, 4 GroumForTestClass.statementList, 3 List.size, 5 Collections.emptyMap, 8 Integer.valueOf, 7 Integer.toString, 6 String.equals, 9 String.valueOf, 10 IF, 13 List.add, 14 FOR, 15 IF, 16 List.addAll],
edges=[1 ArrayList.<init>-->2 List.size
, 4 GroumForTestClass.statementList-->3 List.size
, 2 List.size-->4 GroumForTestClass.statementList
, 3 List.size-->5 Collections.emptyMap
, 8 Integer.valueOf-->7 Integer.toString
, 7 Integer.toString-->6 String.equals
, 5 Collections.emptyMap-->8 Integer.valueOf
, 6 String.equals-->9 String.valueOf
, 9 String.valueOf-->10 IF
, 10 IF-->13 List.add
, 13 List.add-->14 FOR
, 14 FOR-->15 IF
, 15 IF-->16 List.addAll
, 1 ArrayList.<init>-->13 List.add
, 2 List.size-->13 List.add
, 1 ArrayList.<init>-->16 List.addAll
, 2 List.size-->16 List.addAll
, 13 List.add-->16 List.addAll
, 10 IF-->15 IF
]
}
```

**Comment by tsantalis** [comment]
@palashborhanuddin 
Can you please show me in the code how exactly you implemented the following
> Use method return type as Class type for the following method in chain of method invocation
`memcachedWriteInstancesByServerGroup.get(key).get(0)`

**Comment by palashborhanuddin** [comment]
[Here getInvokerVariableType](https://github.com/palashborhanuddin/RefactoringMatcher/commit/144c575e0112b1825ff7fd9b005425188e284b84#diff-9ff9c06641c3a8d8295b9843431f3415e202826c8def2b5c79638b8f7de1216cR568) is the method which is implemented to determine the invoker type of a method. [Here](https://github.com/palashborhanuddin/RefactoringMatcher/commit/144c575e0112b1825ff7fd9b005425188e284b84#diff-9ff9c06641c3a8d8295b9843431f3415e202826c8def2b5c79638b8f7de1216cR591), in case of expression type as MethodInvocation I'm checking if we have determined the return type of method in expression, if Yes then use it as the invoker type (which is Class type of C.m label format for the Groum). In the example test code, I have used a statement as
```JAVA
Integer i = java.lang.Integer.valueOf(1).toString().equals("1");
```
and we get Groum nodes as follows:
```JAVA
6 String.equals
7 Integer.toString
8 Integer.valueOf
```

There is 1 known case yet to be handled for the parameterized type of Collections.
